governator-test-junit/src/main/java/com/netflix/governator/guice/test/AnnotationBasedTestInjectorManager.java

```diff
@@ -38,22 +39,28 @@
 
 public class AnnotationBasedTestInjectorManager {
 
-    private final LifecycleInjector injector;
+    private LifecycleInjector injector;
+    private final TestInjectorCreationMode injectorCreationMode;
     private final List<Object> mocksToReset = new ArrayList<>();
     private final List<Module> modulesForTestClass = new ArrayList<>();
     private final List<Module> overrideModules = new ArrayList<>();
     private final List<Key<?>> spyTargets = new ArrayList<>();
     private final SettableConfig classLevelOverrides = new DefaultSettableConfig();
     private final SettableConfig methodLevelOverrides = new DefaultSettableConfig();
     private final TestPropertyOverrideAnnotationReader testPropertyOverrideAnnotationReader = new TestPropertyOverrideAnnotationReader();
-    private TestCompositeConfig testCompositeConfig;
+    private final TestCompositeConfig testCompositeConfig;
 
     public AnnotationBasedTestInjectorManager(Class<?> classUnderTest) {
+        this.injectorCreationMode = getInjectorCreationModeForAnnotatedClass(classUnderTest);
         inspectModulesForTestClass(classUnderTest);
         inspectMocksForTestClass(classUnderTest);
         inspectSpiesForTargetKeys(Elements.getElements(modulesForTestClass));
         testCompositeConfig = new TestCompositeConfig(classLevelOverrides, methodLevelOverrides);
         overrideModules.add(new ArchaiusTestConfigOverrideModule(testCompositeConfig));
```

Comment:
It should be a harmless addition. I can add checks to see if its necessary or not but I figured that wasn't worth the added complexity.